src/OrleansRuntime/Scheduler/WorkItemGroup.cs

```diff
@@ -440,14 +440,21 @@ public string DumpStatus()
                 sb.Append(this);
                 sb.AppendFormat(". Currently QueuedWorkItems={0}; Total EnQueued={1}; Total processed={2}; Quantum expirations={3}; ",
                     WorkItemCount, totalItemsEnQueued, totalItemsProcessed, quantumExpirations);
-                if (AverageQueueLenght != 0)
+                Utils.SafeExecute(() =>
```

Comment:
If this is unsafe, should we add these strings to a new string builder, then append the new string builder to sb at the end of this, if it does not throw?